fv3core/stencils/delnflux.py

```diff
@@ -86,136 +907,242 @@ def diffusive_damp(
         )
     return fx, fy
 
 
-def compute_no_sg(q, fx2, fy2, nord, damp_c, d2, kstart=0, nk=None, mass=None):
-    # q is input only
+def compute_no_sg(q, fx2, fy2, nord, damp_c, d2, mass=None, nk=None):
+    if max(nord[:]) > 3:
+        raise Exception("nord must be less than 3")
+    if not np.all(n in [0, 2, 3] for n in nord[:]):
+        raise NotImplementedError("nord must have values 0, 2, or 3")
+    nmax = int(max(nord[:]))
     grid = spec.grid
-    nord = int(nord)
-    i1 = grid.is_ - 1 - nord
-    i2 = grid.ie + 1 + nord
-    j1 = grid.js - 1 - nord
-    j2 = grid.je + 1 + nord
+    i1 = grid.is_ - 1 - nmax
+    i2 = grid.ie + 1 + nmax
+    j1 = grid.js - 1 - nmax
+    j2 = grid.je + 1 + nmax
     if nk is None:
-        nk = grid.npz - kstart
-    kslice = slice(kstart, kstart + nk)
-    origin_d2 = (i1, j1, kstart)
+        nk = grid.npz
+    origin_d2 = (i1, j1, 0)
     domain_d2 = (i2 - i1 + 1, j2 - j1 + 1, nk)
+    f1_ny = grid.je - grid.js + 1 + 2 * nmax
+    f1_nx = grid.ie - grid.is_ + 2 + 2 * nmax
+    fx_origin = (grid.is_ - nmax, grid.js - nmax, 0)
+
+    preamble_ax_offsets = axis_offsets(spec.grid, origin_d2, domain_d2)
+    full_ax_offsets = axis_offsets(
+        spec.grid, (grid.isd, grid.jsd, 0), (grid.nid, grid.njd, nk)
+    )
+    fx_ax_offsets = axis_offsets(spec.grid, fx_origin, (f1_nx, f1_ny, nk))
+    fy_ax_offsets = axis_offsets(spec.grid, fx_origin, (f1_nx - 1, f1_ny + 1, nk))
+
     if mass is None:
+        d2_damp = gtscript.stencil(
+            definition=d2_damp_interval,
+            externals={
+                "nord0": nord[0],
+                "nord1": nord[1],
+                "nord2": nord[2],
+                "nord3": nord[3],
+                **preamble_ax_offsets,
+            },
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
+        )
         d2_damp(q, d2, damp_c, origin=origin_d2, domain=domain_d2)
     else:
-        copy_stencil(q, d2, origin=origin_d2, domain=domain_d2)
-
-    if nord > 0:
-        corners.copy_corners_x_stencil(
-            d2, origin=(grid.isd, grid.jsd, kstart), domain=(grid.nid, grid.njd, nk)
+        new_copy_stencil = gtscript.stencil(
+            definition=copy_stencil_interval,
+            externals={
+                "nord0": nord[0],
+                "nord1": nord[1],
+                "nord2": nord[2],
+                "nord3": nord[3],
+                **preamble_ax_offsets,
+            },
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
         )
-    f1_ny = grid.je - grid.js + 1 + 2 * nord
-    f1_nx = grid.ie - grid.is_ + 2 + 2 * nord
-    fx_origin = (grid.is_ - nord, grid.js - nord, kstart)
+        new_copy_stencil(q, d2, origin=origin_d2, domain=domain_d2)
 
-    fx2_order(
-        d2, grid.del6_v, fx2, order=1, origin=fx_origin, domain=(f1_nx, f1_ny, nk)
+    conditional_corner_copy_x = gtscript.stencil(
```

Comment:
I hate to make you spend more time on this incredible PR that already does so much, but this runtime call to gtscript.stencil will cause a slowdown I believe... we may want to make a DelnFlux class, though we could indeed knowingly slow down master and open a new PR to turn it into a class

Reply:
I have to agree, we should avoid calling `gtscript.stencil` directly and opt instead for `StencilWrapper`.